Creatures lair is empty

### What is wrong
Creatures lair is empty when i enter into the mine first time.
It's selffixed after i exit from the mine in Alma and come back after few hours.


| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  release/git | eg. JA2 Gold Dutch  |    yes/no    |    yes/no     |
ja2-stracciatella-0.17.0-1.x86_64 | JA2GOLD Russian

[savegame.tar.gz](https://github.com/ja2-stracciatella/ja2-stracciatella/files/5245693/savegame.tar.gz)
Just load my save game and go to the mine
Linux
connected mods:
"mods": ["from-russia-with-love","imp-quiz-honest-answers"]

![I also went all the way down the mine without a single creature present](https://i.imgur.com/Y1nXIu7.png)

I don't have a before savegame as I'm quicksaving my way through the game.

```
2021-03-20T07:19:51 [ INFO] game/GameLoop.cc: Version Label: Stracciatella v0.17.1-git+866226846
2021-03-20T07:19:51 [ INFO] game/GameLoop.cc: Version #:     Build 04.12.02
2021-03-20T07:20:01 [ WARN] game/Strategic/StrategicMap_Secrets.cc: No secret defined at sector 49
2021-03-20T07:20:01 [ WARN] game/Strategic/StrategicMap_Secrets.cc: No secret defined at sector 49
2021-03-20T07:20:02 [ WARN] game/Strategic/StrategicMap_Secrets.cc: No secret defined at sector 49
2021-03-20T07:20:02 [ WARN] game/Strategic/StrategicMap_Secrets.cc: No secret defined at sector 49
2021-03-20T08:11:07 [ WARN] game/Strategic/Game_Clock.cc: Call to UnPauseGame() while Pause State is LOCKED!
2021-03-20T08:13:34 [ERROR] game/Strategic/Strategic_AI.cc: Attempting to send reinforcements from a garrison that doesn't have any! -- KM:0 (with prior saved game and Debug Log)
2021-03-20T08:16:05 [ERROR] game/Strategic/Strategic_AI.cc: Attempting to send reinforcements from a garrison that doesn't have any! -- KM:0 (with prior saved game and Debug Log)
2021-03-20T08:17:25 [ WARN] game/Strategic/Game_Clock.cc: Call to UnPauseGame() while Pause State is LOCKED!
2021-03-20T08:27:57 [ WARN] game/Tactical/Tactical_Save.cc: SynchronizeItemTempFile() Reported 0, should be 5
2021-03-20T08:44:57 [ WARN] game/Tactical/Tactical_Save.cc: SynchronizeItemTempFile() Reported 0, should be 6
2021-03-20T08:45:20 [ WARN] game/Tactical/Tactical_Save.cc: SynchronizeItemTempFile() Reported 0, should be 6
2021-03-20T08:51:45 [ WARN] game/TileEngine/Structure.cc: Structure is not defined
2021-03-20T08:51:46 [ WARN] game/TileEngine/Structure.cc: Structure is not defined
2021-03-20T08:51:47 [ WARN] game/TileEngine/Structure.cc: Structure is not defined
2021-03-20T08:51:48 [ WARN] game/TileEngine/Structure.cc: Structure is not defined
```

--- a/src/game/Strategic/Auto_Resolve.cc
+++ b/src/game/Strategic/Auto_Resolve.cc
@@ -414,6 +414,8 @@ static void DoTransitionFromPreBattleInterfaceToAutoResolve(void)
 
 void EnterAutoResolveMode(const SGPSector& ubSector)
 {
+	//fix empty creatures' lair bug
+	gubCreatureBattleCode = CREATURE_BATTLE_CODE_NONE;
 	//Set up mapscreen for removal
 	SetPendingNewScreen( AUTORESOLVE_SCREEN );
 	CreateDestroyMapInvButton();

